### ISSUE TITLE
Huge lack of documentation

### ISSUE THREAD

**Issue by AlmightyFloppyFish** [antoyo/relm#134]
I'm starting to learn rust and I love elm, so seeing this library existed made my day. However, there's a huge lack of any kind of documentation. 

The only way i can learn how to use relm is by

1. Looking at the readme / website
2. Looking at the complicated examples
3. Looking at the docs.rs page

The readme and website instructions only gives you a vague idea of how to use relm and the code in the readme doesn't even compile without modifications, which of course, you can't make before knowing Relm.

The other alternatives aren't the best introduction to a new library... To learn something like relm, which also expects a certain program structure and not just has methods, needs a step by step tutorial.

Being able to learn Relm would be amazing for me since after trying Elm I can't imagine using anything else, if i eventually do manage to learn Relm I'll be sure to contribute documentation myself. Just thought I'd let you know that this is a huge problem and I can't imagine Relm grow unless this improves.

TL;DR: I've sat 3 hours and havn't managed to open a blank window using relm yet.

**Comment by antoyo** [comment]
I completely understand that need.
I wanted to release [version 1.0](https://github.com/antoyo/relm/issues/102) before writing the tutorial to avoid having to rewrite it, so that might take more time. Sorry about that, but I'm mostly alone working on this project.

The example in the readme is similar to [this one](https://github.com/antoyo/relm/blob/master/examples/buttons-attribute.rs), so that might help you.

**Comment by antoyo** [comment]
Closing in favor of #26.
